DDDEastAnglia.Tests/Helpers/Email/SessionSubmissionEmailSender_Should.cs

```diff
@@ -0,0 +1,196 @@
+﻿using DDDEastAnglia.Helpers.Email;
+using DDDEastAnglia.Helpers.File;
+using DDDEastAnglia.Models;
+using NSubstitute;
+using NUnit.Framework;
+using System;
+using System.Net.Mail;
+
+namespace DDDEastAnglia.Tests.Helpers.Email
+{
+    [TestFixture]
+    public class SessionSubmissionEmailSender_Should
+    {
+        [Test]
+        public void ThrowAnExceptionWhenConstructed_WhenTheSuppliedEmailSenderIsNull()
+        {
+            var messageFactory = Substitute.For<IMessageFactory>();
+            var fileContentsProvider = Substitute.For<IFileContentsProvider>();
+            Assert.Throws<ArgumentNullException>(() => new SessionSubmissionEmailSender(null, messageFactory, fileContentsProvider));
+        }
+
+        [Test]
+        public void ThrowAnExceptionWhenConstructed_WhenTheSuppliedMessageFactoryIsNull()
+        {
+            var emailSender = Substitute.For<IEmailSender>();
+            var fileContentsProvider = Substitute.For<IFileContentsProvider>();
+            Assert.Throws<ArgumentNullException>(() => new SessionSubmissionEmailSender(emailSender, null, fileContentsProvider));
+        }
+
+        [Test]
+        public void ThrowAnExceptionWhenConstructed_WhenTheSuppliedFileContentsProviderIsNull()
+        {
+            var emailSender = Substitute.For<IEmailSender>();
+            var messageFactory = Substitute.For<IMessageFactory>();
+            Assert.Throws<ArgumentNullException>(() => new SessionSubmissionEmailSender(emailSender, messageFactory, null));
+        }
+
+        [Test]
+        public void SendAnEmailFromTheCorrectEmailAddressAndName()
```

Comment:
With the refactoring I mentioned below, I think these tests would go away as the class wouldn't be doing any sending of emails any more.